src/dali/transaction_resolver.py

```diff
@@ -231,70 +232,90 @@ def resolve_transactions(
-            else:
-                transaction_list: List[AbstractTransaction]
-                transaction_list = unique_id_2_transactions.setdefault(AssetAndUniqueId(transaction.asset, transaction.unique_id), [])
-                transaction_list.append(transaction)
-                unique_id_2_transactions[AssetAndUniqueId(transaction.asset, transaction.unique_id)] = transaction_list
-
-        for transaction_list in unique_id_2_transactions.values():
-            if len(transaction_list) > 2:
-                raise RP2RuntimeError(
-                    f"Internal error: Attempting to resolve more than two transactions with same {Keyword.UNIQUE_ID.value}: {transaction_list}"
-                )
-            if len(transaction_list) == 1:
-                transaction = _apply_transaction_hint(transaction_list[0], global_configuration)
                 if read_spot_price_from_web:
                     transaction = _update_spot_price_from_web(transaction, global_configuration)
-                LOGGER.debug("Self-contained transaction: %s", str(transaction))
-                resolved_transactions.append(transaction)
-                continue
-            if len(transaction_list) == 0:
-                raise RP2RuntimeError(f"Internal error: Attempting to resolve zero transactions: {transaction_list}")
 
-            transaction1: AbstractTransaction = transaction_list[0]
-            transaction2: AbstractTransaction = transaction_list[1]
-
-            if transaction1.unique_id != transaction2.unique_id:
-                raise RP2RuntimeError(
-                    f"Internal error: transaction1.{Keyword.UNIQUE_ID.value} != transaction2.{Keyword.UNIQUE_ID.value}: {transaction1}\n{transaction2}"
-                )
-            if transaction1.asset != transaction2.asset:
-                raise RP2RuntimeError(
-                    f"Internal error: transaction1.{Keyword.ASSET.value} != transaction2.{Keyword.ASSET.value}: {transaction1}\n{transaction2}"
-                )
-
-            if isinstance(transaction1, InTransaction) and isinstance(transaction2, OutTransaction):
-                transaction = _resolve_in_out_transaction(transaction1, transaction2, None)
-            elif isinstance(transaction1, OutTransaction) and isinstance(transaction2, InTransaction):
-                transaction = _resolve_out_in_transaction(transaction1, transaction2, None)
-            elif isinstance(transaction1, IntraTransaction) and isinstance(transaction2, IntraTransaction):
-                transaction = _resolve_intra_intra_transaction(transaction1, transaction2, None)
-            else:
-                raise RP2RuntimeError(
-                    f"Internal error: attempting to resolve two transactions that aren't Intra/Intra, In/Out or Out/In:\n{transaction1}\n{transaction2}"
-                )
-
-            if read_spot_price_from_web:
-                transaction = _update_spot_price_from_web(transaction, global_configuration)
```

Comment:
These two lines should not be removed.

Reply:
I think this is showing up as removed due to indentation changes, viewing with whitespace off shows this being kept.